packages/lib/services/synchronizer/syncInfoUtils.test.ts

```diff
@@ -200,6 +200,34 @@ describe('syncInfoUtils', () => {
 		logger.enabled = true;
 	});
 
+	// cSpell:disable
+	it('should filter unnecessary sync info', async () => {
+		const syncInfo = new SyncInfo();
+		syncInfo.masterKeys = [{
+			id: '1',
+			content: 'longstringverylongstringlongstringverylongstringlongstringverylongstring',
+			checksum: 'longstringverylongstringlongstringverylongstringlongstringverylongstring',
+		}];
+		syncInfo.ppk = {
+			id: '1',
+			publicKey: 'longstringverylongstringlongstringverylongstringlongstringverylongstring',
+			privateKey: {
+				encryptionMethod: 1,
+				ciphertext: 'longstringverylongstringlongstringverylongstringlongstringverylongstring',
+			},
+			createdTime: 0,
+			keySize: 0,
+		};
+		const filteredSyncInfo = syncInfo.filterSyncInfo();
+
+		expect(filteredSyncInfo.masterKeys_[0].content).toBeUndefined();
+		expect(filteredSyncInfo.masterKeys_[0].checksum).toBeUndefined();
+
+		expect(filteredSyncInfo.ppk_.value.publicKey).toBe('longstringverylongstringlongstringverylo');
+		expect(filteredSyncInfo.ppk_.value.privateKey.ciphertext).toBe('longstringverylongst...stringverylongstring');
```

Comment:
To make it clearer what to expect, and to check all properties, could you have instead a single `expect(filteredSyncInfo).toEqual(...)` and that should be a comparison with the full object we should expect

Reply:
Does my approach in https://github.com/laurent22/joplin/pull/10014/commits/441b85ea723c3f0cf61131527d36aca6534e97f4 look better?